### PR TITLE
Update HPKE encryption of match keys

### PR DESCRIPTION
**Pull request by akoshelev**
It addresses the updates in the proposal https://github.com/patcg-individual-drafts/ipa/pull/31/ and IPA issue #492

### CODE REVIEW

File: src/hpke/info.rs [hunk 1]
```diff
@@ -7,16 +7,18 @@ const DOMAIN: &str = "private-attribution";
...
 ///
 /// [`info`]: https://www.rfc-editor.org/rfc/rfc9180.html#name-creating-the-encryption-con
 #[derive(Clone)]
 pub struct Info<'a> {
     pub(super) key_id: KeyIdentifier,
     pub(super) epoch: Epoch,
+    pub(super) match_key_provider_origin: &'a str,
     pub(super) helper_origin: &'a str,
-    pub(super) site_origin: &'a str,
+    pub(super) site_registrable_domain: &'a str,
```
Comment (martinthomson) [inline src/hpke/info.rs:21]:
```suggestion
    pub(super) site_domain: &'a str,
```

This might be easier to type for us.

File: src/hpke/mod.rs [hunk 2]
```diff
@@ -148,8 +148,9 @@ mod tests {
     }
 
     impl<R: RngCore + CryptoRng> EncryptionSuite<R> {
+        const MKP_ORIGIN: &'static str = "";
         const HELPER_ORIGIN: &'static str = "foo";
-        const SITE_ORIGIN: &'static str = "bar";
+        const SRD: &'static str = "xn--mozilla.com.xn--example.com";
```
Comment (martinthomson) [inline src/hpke/mod.rs:153]: "xn--" hah

**Review by martinthomson — APPROVED** [review]

File: src/hpke/mod.rs (hunk 2 above)
Comment (akoshelev) [reply src/hpke/mod.rs:153]: this was my interpretation of this sentence

```
The ASCII serialization of the registrable domain for the current site as a period-separated sequence of
A-labels
```

by changing it I wanted to validate my understanding by you - it seems we have to prefix everything with `xn--` according to the [spec](https://www.rfc-editor.org/rfc/rfc5890#section-2.3.2.5) ?

Comment (martinthomson) [reply src/hpke/mod.rs]: Ah, my error: NR-LDH is also OK.  These are perfectly OK as NR-LDH labels.

**Pull request merged**